crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1307,6 +1307,48 @@ impl<'src> Lexer<'src> {
         }
     }
 
+    /// Re-lex the current token in the context of a logical line.
+    ///
+    /// Returns a boolean indicating that whether the new current token is different than the
+    /// previous current token. This also means that the current position of the lexer has changed
+    /// and the caller is responsible for updating it's state accordingly.
+    ///
+    /// This method is a no-op if the lexer isn't in a parenthesized context.
```

Comment:
I would expand the comment with some explanation for which tokens this is relevant. Like which tokens may change in this situation or what kind of different tokens could be emitted.